src/include/snapper/dialogs.rb

```diff
@@ -343,33 +261,30 @@ def CreateSnapshotPopup(pre_snapshots)
       created
     end
 
-
+    
     # Popup for deleting existing snapshot
     # @return true if snapshot was deleted
-    def DeleteSnapshotPopup(snapshot)
-      snapshot = deep_copy(snapshot)
-      num = Ops.get_integer(snapshot, "num", 0)
-      pre_num = Ops.get_integer(snapshot, "pre_num", num)
-      type = Ops.get_symbol(snapshot, "type", :none)
-
-      if type != :POST
-
-        # yes/no popup question
-        if Popup.YesNo(_("Really delete snapshot %{num}?") % { :num => num })
-          return Snapper.DeleteSnapshot([ num ])
+    def DeleteSnapshotPopup(snapshots)
+      snaps = []
+      post_snaps = []
+      snapshots.map do |s|
+        if s["type"] != :POST
+          snaps << s["num"]  
+        else
+          post_snaps << {:pre_num => s["pre_num"], :num => s["num"]}
         end
-
-      else
-
-        # yes/no popup question
-        if Popup.YesNo(_("Really delete snapshots %{pre} and %{post}?") %
-                       { :pre => pre_num, :post => num })
-          return Snapper.DeleteSnapshot([ pre_num, num ])
+      end
+      
+      nums = snaps + post_snaps.map {|k| "#{k[:pre_num]}&#{k[:num]}"}
+
+      # yes/no popup question
+      if Popup.YesNo(_("You have selected those snapshots: %{nums}" % { :nums => nums.join(",") } ))
```

Comment:
wrong usage of parens. it should be `_("You have selected those snapshots: %{nums}") % { :nums => nums.join(",") } ` as you have to do first translation and then replacement for `%{nums}` otherwise gettext won't match text.